### DISCUSSION TITLE
ACL Syntax - Allow users to subscribe to a topic based on their IP address

### DISCUSSION THREAD

**Discussion by waqasahmed-uk** [emqx/emqx#12704] [Q&A]
I am trying to find a way where users can subscribe only to a topic which is named the same as their IP address. So for example a user with an IP address of 10.10.10.1 can only subscribe to the topic a/10.10/10/1

The documentation shows how to input this into the ACL file, but instead of a long list of IP addresss for every single device and topic, I was hoping their was some syntax which can be used to easily check this for all devices and topics.

The syntax for the ACL file is below, but this would need to be manually entered for every single topic and device:

%% Allow users with IP address "127.0.0.1" to publish/subscribe to topics "$SYS/#", "#"
{allow, {ipaddr, "127.0.0.1"}, all, ["$SYS/#", "#"]}.

**Comment by ieQu1** [comment]
I'm not sure if this use case is supported by the file-based ACL right now. 
Is it possible to use client id or username instead of the IP address? Both client ID and username can be used as template variables in the topic names.

**Reply by waqasahmed-uk** [reply]
The issue with using client ID or username is that we would manually need to enter them into a list (eg: allow UserA access to Topic1) because in order to keep things secure doing it this way would mean randomly generated Client ID's and Topics. The Client ID's can be changed by a rogue client whereas the IP addresses (all on our internal network cannot be spoofed so I was hoping that tieing IP address to Topic access was going to be a more secure way of doing things. If we were to go down the path of ACL to allow ClientID access to a topic of the same name as the clientID, is there a syntax we can use to set up the control for all clients using one line, instead of adding them all into a long list?

**Comment by zmstone** [comment]
You’ll need to implement a plugin for this 

https://github.com/emqx/emqx-plugin-template/blob/192a43f71d1daf1f8dcff6c7410ff6c115072d30/src/emqx_plugin_template.erl#L111
